Paws4Me/Paws4Me/Controllers/Logic/SignInViewController.swift

```diff
@@ -0,0 +1,142 @@
+            errorMessage = "Incorrect Username"
+            addErrorBorderSingle(txtUsername, txtPassword, nameTextField: "Username")
+            break breakIf
+        } else if !bValidPassword {
+            errorMessage = "Incorrect Password"
+            addErrorBorderSingle(txtPassword, txtPassword, nameTextField: "Password")
+            break breakIf
+        }
+        displayErrorAlert(alertTitle: "Invalid credentials.",
+                                            alertMessage: errorMessage,
+                                            alertActionTitle: "Try again" ,
+                                            alertDelegate: self)
+            }
+            }
+
+func resetBorderSingle(_ textField1: UITextField, bottomLine: CALayer) {
+    bottomLine.backgroundColor = UIColor.init(red: 55/255, green: 21/255, blue: 67/255, alpha: 1).cgColor
+    bottomLine.frame = CGRect(x: 0, y: textField1.frame.height * 1.2, width: textField1.frame.width, height: 2)
+}
+
+func addErrorBorderSingle (_ textField1: UITextField, _ textField2: UITextField, nameTextField: String) {
+    if nameTextField == "Username" {
+        bottomLine.backgroundColor = (UIColor.red).cgColor
+    } else {
+        resetBorderSingle(textField1, bottomLine: bottomLine)
+    }
+    textField1.borderStyle = .none
+    textField1.layer.addSublayer(bottomLine2)
+    if nameTextField == "Password" {
+        bottomLine2.backgroundColor = (UIColor.red).cgColor
+    } else {
+        resetBorderSingle(textField1, bottomLine: bottomLine2)
+    }
+    textField2.borderStyle = .none
+    textField2.layer.addSublayer(bottomLine2)
+}
+
+func addErrorBorderBoth (_ textField1: UITextField, _ textField2: UITextField) {
+    bottomLine.frame = CGRect(x: 0, y: textField1.frame.height * 1.2, width: textField1.frame.width, height: 2)
+    bottomLine2.frame = CGRect(x: 0, y: textField2.frame.height * 1.2, width: textField2.frame.width, height: 2)
+    bottomLine.backgroundColor = (UIColor.red).cgColor
+    bottomLine2.backgroundColor = (UIColor.red).cgColor
+    textField1.borderStyle = .none
+    textField2.borderStyle = .none
+    textField1.layer.addSublayer(bottomLine)
+    textField2.layer.addSublayer(bottomLine2)
+}
+
+extension UIViewController {
+
+    func displayErrorAlert(alertTitle: String,
+                           alertMessage: String,
+                           alertActionTitle: String,
+                           alertDelegate: UIViewController) {
+        let alert = UIAlertController(title: alertTitle,
+                                      message: alertMessage,
+                                      preferredStyle: UIAlertController.Style.alert)
+        alert.addAction(UIAlertAction(title: alertActionTitle, style: UIAlertAction.Style.default, handler: nil))
+        alertDelegate.present(alert, animated: true, completion: nil)
+    }
+}
+
+extension UIViewController {
+    func addCornerRadius (_ button: UIButton) {
+        button.layer.cornerRadius = 10
```

Comment:
This can also be added to the style kit and the button styling can rather be added to a `UIButton` `extension`

Reply:
Noted 👍🏻